src/cedalion/nirs.py

```diff
@@ -163,6 +163,44 @@ def od2conc(
 
     return conc
 
+def conc2od(
+    conc: xr.DataArray,
+    geo3d: xr.DataArray,
+    dpf: xr.DataArray,
+    spectrum: str = "prahl",
+):
+    """Calculate optical density data from concentration changes.
+
+    Args:
+        conc (xr.DataArray, (channel, *)): The concentration changes by channel.
+        geo3d (xr.DataArray): The 3D coordinates of the optodes.
+        dpf (xr.DataArray, (wavelength, *)): The differential pathlength factor data.
+        spectrum (str, optional): The type of spectrum to use for calculating extinction
+            coefficients. Defaults to "prahl".
+
+    Returns:
+        od (xr.DataArray, (channel, wavelength, *)): A data array containing
+            optical density data.
+    """
+
+    conc = conc.pint.to("molar")
+
+    # Get the extinction coefficients for the chosen spectrum
+    wavelengths = dpf.wavelength.values.astype(float)
+    E = cedalion.nirs.get_extinction_coefficients(spectrum, wavelengths)
```

Comment:
Previously, this used hard-coded wavelengths. We have the original wavelengths in dpf.wavelengths, so use these.